packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -306,6 +321,33 @@ EmbedFlutterFrameworks() {
     RunCommand codesign --force --verbose --sign "${EXPANDED_CODE_SIGN_IDENTITY}" -- "${xcode_frameworks_dir}/App.framework/App"
     RunCommand codesign --force --verbose --sign "${EXPANDED_CODE_SIGN_IDENTITY}" -- "${xcode_frameworks_dir}/Flutter.framework/Flutter"
   fi
+
+  AddObservatoryBonjourService
+}
+
+# Add the observatory publisher Bonjour service to the produced app bundle Info.plist.
+AddObservatoryBonjourService() {
+  ParseFlutterBuildMode
+  local build_mode="${retval}"
+  # Debug and profile only.
+  if [[ "${build_mode}" == "release" ]]; then
+    return
+  fi
+  local built_products_plist="${BUILT_PRODUCTS_DIR}/${INFOPLIST_PATH}"
+
+  # If there are already NSBonjourServices specified by the app (uncommon), insert the observatory service name to the existing list.
+  if plutil -extract NSBonjourServices xml1 -o - "${built_products_plist}"; then
+    RunCommand plutil -insert NSBonjourServices.0 -string "_dartobservatory._tcp" "${built_products_plist}"
+  else
+    # Otherwise, add the NSBonjourServices key and observatory service name.
+    RunCommand plutil -insert NSBonjourServices -json "[\"_dartobservatory._tcp\"]" "${built_products_plist}"
+  fi
+
+  # Don't override the local network description the Flutter app developer specified (uncommon).
+  # This text will appear below the "Your app would like to find and connect to devices on your local network" permissions popup.
+  if ! plutil -extract NSLocalNetworkUsageDescription xml1 -o - "${built_products_plist}"; then
```

Comment:
Display an error message otherwise?

Reply:
If the developer sets their own NSLocalNetworkUsageDescription because their app is doing their own local networking we don't override it.

Reply:
that's ok. I just meant if that's the case, we should print something in the console to indicate that you'll get a popup in debug and we can't set the message in it but it's for flutter tools to connect.